### PR TITLE
Main window/title

### PR DESCRIPTION
**Pull request by zaryabkhan-rs**
> ### Motivate of the pull request
> - [x] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
Foedag's Main Window title was showed in this pattern:  "FOEDAG - current opened file path".
 Screenshot: 
image

> #### What is currently done? (Provide issue link if applicable)
Now main window title is changed to following consistent pattern:  "current-file-open-in-editor" - "project-name" - FOEDAG.
>
> #### What does this pull request change?
It will change the title pattern of main window. 
> #### Test cases:
Title on the welcome page: "FOEDAG"
Title when project created,  Project Name without open any file in editor : <project_name> - "FOEDAG"
Title when open source file in editor : <file_name> - <Project_name> - "FOEDAG"

> #### Testing Steps:
>- Run the Gui and create or open new project.
>- Check the title on the window after creating or opening project.
>- Main title shows in the following format that is shown in the image below

 Screenshot:
image
 Screenshot:
image
>- After closing the any tab, Title will set to the current tab opened in the editor
>- If No tab is opened, Window's title will set to default in this format-> "project_name" - FOEDAG 
 Screenshot:
image

> ### Which part of the code base require a change
> 
> - [x] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [x] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -183,7 +183,12 @@ void MainWindow::openFileSlot() {
 }
 
 void MainWindow::newDesignCreated(const QString& design) {
-  setWindowTitle(m_projectInfo.name + " - " + design);
+  const QFileInfo path(design);
+  if (design.isEmpty())
+    setWindowTitle(m_projectInfo.name);
+  else
+    setWindowTitle(path.baseName() + " - " + m_projectInfo.name);
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:190]: where is project name?

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -183,7 +183,12 @@ void MainWindow::openFileSlot() {
 }
 
 void MainWindow::newDesignCreated(const QString& design) {
-  setWindowTitle(m_projectInfo.name + " - " + design);
+  const QFileInfo path(design);
+  if (design.isEmpty())
+    setWindowTitle(m_projectInfo.name);
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:188]: please create one method which will contain all logic: void SetWindowTitle(const QString& filename, const QString& project, const QString& projectInfo);

File: src/MainWindow/main_window.cpp [hunk 3]
```diff
@@ -688,6 +695,17 @@ void MainWindow::newDialogAccepted() {
   ReShowWindow(strproject);
 }
 
+void MainWindow::slotTabChanged(int index) {
+  QString strName = TextEditorForm::Instance()->GetTabWidget()->tabText(index);
+
+  if (index == -1)
+    setWindowTitle(m_projectManager->getProjectName() + " - " +
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:702]: use SetWindowTitle

File: src/MainWindow/main_window.h [hunk 4]
```diff
@@ -26,8 +26,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "Main/AboutWidget.h"
 #include "NewProject/new_project_dialog.h"
+#include "TextEditor/text_editor_form.h"
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.h:29]: Please move to *.cpp

**Review by ravic-rs — CHANGES_REQUESTED** [review]
Please address Volodymyr's comments. Also, why didn't we catch the build error with text_editor_form.h locally ?

File: src/MainWindow/main_window.cpp [hunk 5]
```diff
@@ -137,6 +137,17 @@ void MainWindow::Info(const ProjectInfo& info) { m_projectInfo = info; }
 
 ProjectInfo MainWindow::Info() const { return m_projectInfo; }
 
+void MainWindow::SetWindowTitle(const QString& filename, const QString& project,
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:140]: @KochynVolodymyr do we need this anymore ?
https://github.com/os-fpga/FOEDAG/pull/639/files#diff-9e31125b4e2b6bd449e09a6b478450cb54bf4fb9ddb23baadb06ad5d9a9bb149R63

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:140]: What do you mean?

Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:140]: @zaryabkhan-rs do we need these lines ? What's the behavior when we have an empty project ?

MainWindow::MainWindow(Session* session) : m_session(session) {
  /* Window settings */
  setWindowTitle(tr("FOEDAG"));

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**